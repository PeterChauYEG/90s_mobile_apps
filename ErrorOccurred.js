/**
 * @format
 * @flow
 */

// react
import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

// react native
import { Button, Text } from 'react-native'

// styles 
import styles from './styles/ErrorOccurred'

const ErrorOccurred = props => {
  const { clearLyrics } = props
  
  return (
    <Fragment>
      <Text style={styles.title}>Uh oh, an error has occurred</Text>    
      <Button
        accessibilityLabel='Start this process over and enter new parameters'
        color='#841584'
        onPress={() => clearLyrics()}
        title='Start over'
      /> 
    </Fragment>
  )
}

ErrorOccurred.propTypes = {
  clearLyrics: PropTypes.func.isRequired
}
ErrorOccurred.defaultProps = {}

export default ErrorOccurred