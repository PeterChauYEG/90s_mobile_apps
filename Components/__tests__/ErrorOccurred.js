import React from 'react'
// import { shallow } from 'enzyme'
import ErrorOccurred from '../ErrorOccurred'
import renderer from 'react-test-renderer'

describe('ErrorOccurred', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ErrorOccurred clearLyrics={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('simulates "Start over" button click', () => {
  //   const clearLyrics = jest.fn()
  //
  //   const wrapper = shallow(
  //     <ErrorOccurred clearLyrics={clearLyrics}
  //     />
  //   )
  //
  //   wrapper.find('Button').simulate('click')
  //
  //   expect(clearLyrics).toHaveBeenCalled()
  // })
})
