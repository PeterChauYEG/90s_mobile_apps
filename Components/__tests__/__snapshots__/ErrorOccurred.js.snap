// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorOccurred renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Uh oh, an error has occurred
  </Text>,
  <View
    accessibilityLabel="Start this process over and enter new parameters"
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }
      >
        Start over
      </Text>
    </View>
  </View>,
]
`;
